### PR TITLE
TDEngine Trace archiver

### PR DESCRIPTION
**Pull request by Modularius**

### CODE REVIEW

**Review by DanNixon — CHANGES_REQUESTED** [review]
This is from an initial look, I haven't looked at all the files yet but the style comments apply to pretty much everything.

I think the `docker-compose` strategy needs looking at, it does not really fit well with the rest of the project. I think assuming Redpanda is available is fine, include documentation on starting TDengine via Podman, and assume the archiver will be run (for development at least) via `cargo run`.

File: Cargo.toml [hunk 1]
```diff
@@ -3,11 +3,11 @@ resolver = "2"
 members = [
   "common",
   "events-to-histogram",
-  "kafka-daq-report",
```
[the remark this answers is not in the document]
Comment (Modularius) [reply Cargo.toml:6]: This has been corrected now

**Comment by Modularius** [comment]
I think I missed a few, it'll be ready the next commit though

**Comment by DanNixon** [comment]
I still see `kafka-daq-report` removed.

**Review by DanNixon — APPROVED** [review]

**Pull request merged**